src/factoid/responder.rs

```diff
@@ -0,0 +1,100 @@
+use super::util;
+use std::boxed::Box;
+use util::is_extension;
+
+pub struct Responders {
+    responders: Vec<Box<dyn Responder>>,
+}
+
+impl Responders {
+    pub fn new() -> Responders {
+        Responders {
+            responders: Vec::<Box<dyn Responder>>::new(),
+        }
+    }
+
+    pub fn register_responder<T: 'static + Responder>(&mut self, responder: T) {
+        self.responders.push(Box::new(responder));
+    }
+
+    pub fn respond(&self, input: &String) -> Vec<String> {
```

Comment:
it's generally nicer to work with `&str` than `&String`, though they're very similar. notable difference, `"string literals"` are `&'static str` and don't require conversion, where going from `String` -> `&str` is `foo.as_str()` (also written as `&**foo` if you like symbols and deref coercion)